packages/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -43,23 +46,37 @@ const DataGridPremiumRaw = React.forwardRef(function DataGridPremium<R extends G
 
   useLicenseVerifier('x-data-grid-premium', releaseInfo);
 
+  const { pivotParams } = props;
+
+  const hasPivotModel = pivotParams?.pivotModel != null;
+
   if (process.env.NODE_ENV !== 'production') {
     validateProps(props, dataGridPremiumPropValidators);
   }
   return (
     <GridContextProvider privateApiRef={privateApiRef} props={props}>
       <GridRoot
-        className={props.className}
+        className={clsx(props.className, hasPivotModel && gridClasses.sidePanel)}
         style={props.style}
         sx={props.sx}
         ref={ref}
         {...props.forwardedProps}
       >
-        <GridHeader />
-        <GridBody>
-          <Watermark packageName="x-data-grid-premium" releaseInfo={releaseInfo} />
-        </GridBody>
-        <GridFooterPlaceholder />
+        <div
+          style={{ display: 'flex', flexDirection: 'column', overflow: 'hidden', flex: 1 }}
+          role="presentation"
+        >
```

Comment:
It's a static style object, could be extracted.

And I'm not sure I love inline styles but I also don't love any of our styling methods :|